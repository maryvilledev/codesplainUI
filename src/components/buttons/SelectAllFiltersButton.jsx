import React, { PropTypes } from 'react';
import { RaisedButton } from 'material-ui';

const SelectAllFiltersButton = ({ selectCallback }) => (
  <RaisedButton
    label="Select All Filters"
    onTouchTap={selectCallback}
    primary
  />
);

SelectAllFiltersButton.propTypes = {
  selectCallback: PropTypes.func.isRequired,
};

export default SelectAllFiltersButton;
