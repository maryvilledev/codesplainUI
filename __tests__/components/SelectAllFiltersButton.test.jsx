import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import SelectAllFiltersButton from '../../src/components/buttons/SelectAllFiltersButton';

describe('<SelectAllFiltersButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <SelectAllFiltersButton
        selectCallback={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: selectCallback', () => {
    it('is triggered when button is clicked', () => {
      const selectCallback = jest.fn();
      const wrapper = shallowWithContext(
        <SelectAllFiltersButton
          selectCallback={selectCallback}
        />
      );
      const button = wrapper.find('RaisedButton');
      button.simulate('touchTap');
      expect(selectCallback).toHaveBeenCalled();
    });
  });
});
