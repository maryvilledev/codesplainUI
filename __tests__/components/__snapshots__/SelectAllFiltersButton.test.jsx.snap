exports[`<SelectAllFiltersButton /> matches snapshot 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={true}
  label="Select All"
  labelPosition="after"
  labelStyle={
    Object {
      "fontSize": "12px",
    }
  }
  onTouchTap={[Function]}
  primary={true}
  secondary={false} />
`;
